### PR TITLE
Add named list accessor

### PR DESCRIPTION
**Pull request by mruoss**
This is a very handy little tool when dealing with k8s resources and I think id would make a nice enhancement to this library. It implements an [accessor](https://hexdocs.pm/elixir/Access.html) that simplifies access to lists with named items like the list of containers, env, secrets, etc.

To get a specific env variable in a deployment, you would just use get_in():

```elixir
get_in(resource, ["spec", "template", "spec", "containers", NamedList.access("my-container"), "env", NamedList.access("some_env")])
```

### CODE REVIEW

**Comment by coryodaniel** [comment]
This is very cool! I didnt know about `Access`!

File: lib/k8s/resource/named_list.ex [hunk 1]
```diff
@@ -0,0 +1,49 @@
+defmodule K8s.Resource.NamedList do
+  @moduledoc """
+  Provides an accessor to a list of maps whereas each element in the list nas a key named "name". The name should
```
Comment (coryodaniel) [inline lib/k8s/resource/named_list.ex:3]:
```suggestion
  Provides an accessor to a list of maps whereas each element in the list has a key named "name". The name should
```

File: lib/k8s/resource/named_list.ex [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    iex> pop_in([%{"name" => "key1", "value" => "value1"}, %{"name" => "key2", "value" => "value2"}], [K8s.Resource.NamedList.access("key1"), "value"])
+    {"value1", [%{"name" => "key1"}, %{"name" => "key2", "value" => "value2"}]}
+  """
+
+  @spec access(binary) :: Access.access_fun(data :: list(), get_value :: term)
+  def access(name) do
+    fn op, data, next ->
+      if Enum.count(data, match_name_callback(name)) > 1 do
+        raise ArgumentError, "The name #{name} ist not unique in the given list: #{inspect(data)}"
```
Comment (coryodaniel) [inline lib/k8s/resource/named_list.ex:22]:
```suggestion
        raise ArgumentError, "The name #{name} is not unique in the given list: #{inspect(data)}"
```

**Review by coryodaniel — APPROVED** [review]
Very cool. Is it common to raise errors when using `Access` vs returning an error tuple? I assume so since the error tuple would blow up any `get_in/` call.

**Comment by coryodaniel** [comment]
Could you add an entry in the changelog!

**Comment by mruoss** [comment]
> Very cool. Is it common to raise errors when using `Access` vs returning an error tuple? I assume so since the error tuple would blow up any `get_in/` call.

Looking at the code of [Access](https://github.com/elixir-lang/elixir/blob/v1.10.4/lib/elixir/lib/access.ex#L779) itself, it seems to be the way to go. And probably that is exactly the reason.

**Review by coryodaniel — APPROVED** [review]
Awesome thanks so much this is really cool.

I’m gonna be cutting a 0.5.X release tomorrow

**Pull request merged**